core/state_prefetcher.go

```
@@ -68,7 +70,13 @@ func (p *statePrefetcher) Prefetch(block *types.Block, statedb *state.StateDB, c
 			return // Also invalid block, bail out
 		}
 		statedb.Prepare(tx.Hash(), i)
-		if err := precacheTransaction(msg, p.config, gaspool, statedb, header, evm); err != nil {
+
+		l1DataFee, err := fees.CalculateL1DataFee(tx, statedb)
+		if err != nil {
+			return
+		}
```

If the only possible error here is from the RLP encoding and if that will not happen in practice, then we could refactor these and panic instead of returning an error. Can be refactored in another PR.